scipy/special/_basic.py

```diff
@@ -2953,6 +2956,139 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number of ways to
+        partition a set with N elements into K non-empty subsets.
+
+        The values this function returns are calculated using a dynamic
+        program which avoids redundant computation across the subproblems
+        in the solution. For array-like input, this implementation also 
+        avoids redundant computation across the different Stirling number
+        calculations.
+
+        The numbers are sometimes denoted
+        .. math::
+
+            `{N \brace K}`;
+
+        see [GKP]_ for details. This is often expressed-verbally-as
+        "N subset K".
+
+        Parameters
+        ----------
+        N : int, ndarray
+            Number of things.
+        K : int, ndarray
+            Number of non-empty subsets taken.
+        exact : bool, optional
+            For integers, if `exact` is False, then floating point precision is
+            used, otherwise the result is computed exactly. For non-integers,
+            if `exact` is True, is truncated. Defaults to True.
```

Comment:
IMO, we should not advertise something that is not implemented yet. E.g. it could say:
```suggestion
        exact : bool, optional
            This keyword is reserved for a planned future implementation
            that allows trading speed for accuracy.
```